src/audio/smart_amp/Kconfig

```diff
@@ -1,19 +1,39 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
-menu "Smart amplifier solutions"
-	visible if COMP_SMART_AMP
+if COMP_SMART_AMP
+
+choice
+        prompt "Smart Amplifier solution applied"
+        default PASSTHRU_AMP
+	help
+	  The selection for Smart Amplifier component implementation
+	  will depend on the Amplifier solution supplier. It is fair
+	  to treat the supported solutions as mutually exclusive ones.
+	  There should be no more than one solution selected per build
```

Comment:
Why can't we setup a system similar to modules and have support for all devices at once? You could also do composition and just have this by a stand in for default implementation in a module?

Reply:
We are able to do it for sure. The code implementation will be a bit more complicated (as how module_adapter is implemented https://github.com/thesofproject/sof/blob/main/src/include/sof/audio/module_adapter/module/generic.h

I didn't implement that way due to the proposed plan in #7801 said the long-term goal is to use module API for all components, so I would defer that to the future task for converting smart_amp to use module API.